README.md
[![Build Status](https://travis-ci.org/psyreactor/jboss-cookbook.svg?branch=master)](https://travis-ci.org/psyreactor/jboss-cookbook)
Jboss Cookbook
==================

This cookbook installs JBoss EAP from a zip. This cookbook is based on [jboss-eap cookbook](https://github.com/andrewfraley/chef-jboss-eap)(not is a fork)

####[Jboss EAP](http://redhat.com/products/jbossenterprisemiddleware/)
"Boss is a division of Red Hat that provides support for the JBoss open source application server program and related services marketed under the JBoss Enterprise Middleware Suite (JEMS) brand. It is an open source alternative to commercial offerings from IBM WebSphere, Oracle BEA Services, and SAP NetWeaver.

The JBoss applications server is a J2EE platform for developing and deploying enterprise Java applications, Web applications and services, and portals.  J2EE allows the use of standardized modular components and enables the Java platform to handle many aspects of programming automatically"

Requirements
------------
- `java` - Not managed by this cookbook, yuo can use [java-cookbook](https://github.com/socrata-cookbooks/java)
- `JBoss EAP 6` - Packaged as a zip and stored on a web server acccessible by the node.
- `JBoss EAP 5` - Packaged as a zip and stored on a web server acccessible by the node.
- 
#### Cookbooks:
- No depends

The following platforms and versions are tested and supported using Opscode's test-kitchen.

- CentOS 5.8, 6.3

The following platform families are supported in the code, and are assumed to work based on the successful testing CentOS.


- Red Hat (rhel)
- Fedora
- Amazon Linux

Recipes
-------
#### jboss:default
The recipe install jboss-eap, according to the version defined in node[:jboss][: version]. 
The cookbook install versions 5 or 6 jboss-eap.

#### jboss:jboss5 or jboss:jboss6 
##### Basic Config
<table>
  <tr>
    <th>Key</th>
    <th>Type</th>
    <th>Description</th>
    <th>Default</th>
  </tr>
  <tr>
    <td><tt>node[:jboss][:version]</tt></td>
    <td>String</td>
    <td>version of jboss to install</td>
    <td><tt>6.2.0</tt></td>
  </tr>
  <tr>
    <td><tt>node[:jboss][:url]</tt></td>
    <td>String</td>
    <td>url for local repo</td>
    <td><tt>http://localhost/jboss/</tt></td>
  </tr>
  <tr>
    <td><tt>node[:jboss][:path]</tt></td>
    <td>String</td>
    <td>Install Path for jboss</td>
    <td><tt>/opt</tt></td>
  </tr>
  <tr>
    <td><tt>node[:jboss][:application]</tt></td>
    <td>String</td>
    <td>name of application</td>
    <td><tt>app</tt></td>
  </tr>
  <tr>
    <td><tt>node[:jboss][:checksum]</tt></td>
    <td>String</td>
    <td>checksum of zip file</td>
    <td><tt>73c542c2e7f1102a3b51ab62e14023bcda227e737233327d2f17aa361c9ff05c</tt></td>
  </tr>
  <tr>
    <td><tt>node[:jboss][:jboss_user]</tt></td>
    <td>String</td>
    <td>user for run jboss</td>
    <td><tt>jboss</tt></td>
  </tr>
  <tr>
    <td><tt>node[:jboss][:jboss_group]</tt></td>
    <td>String</td>
    <td>user group for run jboss</td>
    <td><tt>jboss</tt></td>
  </tr>
  <tr>
    <td><tt>node[:jboss][:java_home]</tt></td>
    <td>String</td>
    <td>set home of java</td>
    <td><tt>nil</tt></td>
  </tr>
  <tr>
    <td><tt>node[:jboss][:java_opts]</tt></td>
    <td>String</td>
    <td>java ops settings</td>
    <td><tt>'-Xms1303m -Xmx1303m -XX:MaxPermSize=256m -Djava.net.preferIPv4Stack=true'</tt></td>
  <tr>
    <td><tt>node[:jboss][:admin_user]</tt></td>
    <td>String</td>
    <td>User for admin console</td>
    <td><tt>admin</tt></td>
  </tr>
  <tr>
    <td><tt>node[:jboss][:admin_passwd]</tt></td>
    <td>String</td>
    <td>Password for admin console user</td>
    <td><tt>password</tt></td>
  </tr>
  <tr>
    <td><tt>node[:jboss][:startup_wait]</tt></td>
    <td>integer</td>
    <td>startup time wait the init script to give error</td>
    <td><tt>30</tt></td>
  </tr>
  <tr>
    <td><tt>node[:jboss][:shutdown_wait]</tt></td>
    <td>integer</td>
    <td>shutdown time wait the init script to give error</td>
    <td><tt>30</tt></td>
  </tr>
  <tr>
    <td><tt>node[:jboss][:port_offset]</tt></td>
    <td>integer</td>
    <td>define port bind offset</td>
    <td><tt>0</tt></td>
  </tr>
</table>


Usage
-----
### jboss::default
#### Example role:

```ruby
name "jboss"
description "JBoss EAP install"
run_list [
    "recipe[jboss]",
    ]

"default_attributes": {
  "jboss": {
      "install_path": "/opt",
      "application": "server1",
      "url": "https://yourserver.local/jboss/",
      "checksum": "0ef5d62a660fea46e0c204a9f9f35ad4",
      "version": "6.2.0",
      "admin_user": "admin",
      "admin_passwd": "password",
      "port_offset": "100"
    }
}

```
Contributing
------------

1. Fork the repository on Github
2. Create a named feature branch (like `add_component_x`)
3. Write your change
4. Write tests for your change (if applicable)
5. Run the tests, ensuring they all pass
6. Submit a Pull Request using Github

[More details](https://github.com/psyreactor/jboss-cookbook/blob/master/CONTRIBUTING.md)

License and Authors
-------------------
Authors:
Lucas Mariani (Psyreactor)
- [marianiluca@gmail.com](mailto:marianiluca@gmail.com)
- [https://github.com/psyreactor](https://github.com/psyreactor)
-------------------
Authors: https://github.com/andrewfraley


CHANGELOG.md
# 0.1.0

First release of Jboss cookbook

* Features
 * Install jboss-eap6
 
# 0.2.0

Second release of Jboss cookbook

* Fixed
 * Correct Config file

# 1.2.0

First Mayor release of Jboss cookbook

* Features
 * add support to jboss-eap5
 * default recipe install jboss correct version
 * create recipe jboss5 and jboss6



spec/centos/jboss5_spec.rb
require_relative '../spec_helper'

describe 'jboss::jboss5 on Centos 6.5' do
  let(:chef_run) do
    ChefSpec::Runner.new(
      :platform => 'centos',
      :version => '6.5'
      ) do |node|
      node.set[:jboss][:version] = '5.2.0'
    end.converge('jboss::jboss5')
  end

  let(:jboss_init) { chef_run.template('/etc/init.d/app') }

  before do
    stub_command('test -h /etc/jboss-as/app').and_return(false)
    stub_command('test -h /var/log/jboss-as/app').and_return(true)
    stub_command('test -h /opt/jboss-as/server/app/log').and_return(false)
  end

  it 'installs a unzip package' do
    expect(chef_run).to install_package('install_unzip')
  end

  it 'creates a user for jboss' do
    expect(chef_run).to create_user('jboss')
  end

  it 'creates a group for jboss user' do
    expect(chef_run).to create_group('jboss')
  end

  it 'download a jboss package' do
    expect(chef_run).to create_remote_file("#{Chef::Config[:file_cache_path]}/jboss-eap-5.2.0.zip")
  end

  it 'extract jboss' do
    expect(chef_run).to run_execute('jboss_extract')
  end

  it 'move directory to detination path' do
    expect(chef_run).to run_execute('jboss_install_path')
  end

  it 'copy profile directory to detination path' do
    expect(chef_run).to run_execute('create_profile')
  end

  it 'change owner jboss' do
    expect(chef_run).to run_execute('jboss_owner')
  end

  it 'creates config directory for jboss' do
    expect(chef_run).to create_directory('/etc/jboss-as')
  end

  it 'creates pid directory for jboss' do
    expect(chef_run).to create_directory('/var/run/jboss-as')
  end

  it 'created jboss config file' do
    expect(chef_run).to create_template('/opt/jboss-as/server/app/conf/run.conf')
  end

  it 'delete symlink to log jboss config file' do
    expect(chef_run).to delete_link('/var/log/jboss-as/app')
  end

  it 'creates log directory for jboss' do
    expect(chef_run).to create_directory('/var/log/jboss-as/app')
  end

  it 'delete default log directory for jboss' do
    expect(chef_run).to delete_directory('/opt/jboss-as/server/app/log')
  end

  it 'create symlink to log jboss config file' do
    expect(chef_run).to create_link('/opt/jboss-as/server/app/log')
  end

  it 'create symlink to config file' do
    expect(chef_run).to create_link('/etc/jboss-as/app')
  end

  it 'declare jboss service' do
    execute = chef_run.service('app')
    expect(execute).to do_nothing
  end

  it 'create service script jboss' do
    expect(chef_run).to create_template('/etc/init.d/app')
  end

  it 'sends a notification to services' do
    expect(jboss_init).to notify('service[app]').to(:enable)
    expect(jboss_init).to notify('service[app]').to(:start)
  end

end


spec/centos/jboss6_spec.rb
require_relative '../spec_helper'

describe 'jboss::jboss6 on Centos 6.5' do
  let(:chef_run) do
    ChefSpec::Runner.new(
      :platform => 'centos',
      :version => '6.5'
      ) do |node|
      node.set[:jboss][:version] = '6.2.0'
    end.converge('jboss::jboss6')
  end

  let(:jboss_init) { chef_run.template('/etc/init.d/app') }
  let(:jboss_standalone) { chef_run.template('/opt/app/bin/standalone.conf') }

  before do
    stub_command('test -h /var/log/jboss-as/app').and_return(true)
    stub_command('test -h /opt/app/standalone/log').and_return(false)
    stub_command('grep admin /opt/app/standalone/configuration/mgmt-users.properties').and_return(false)
  end

  it 'installs a unzip package' do
    expect(chef_run).to install_package('install_unzip')
  end

  it 'creates a user for jboss' do
    expect(chef_run).to create_user('jboss')
  end

  it 'creates a group for jboss user' do
    expect(chef_run).to create_group('jboss')
  end

  it 'download a jboss package' do
    expect(chef_run).to create_remote_file("#{Chef::Config[:file_cache_path]}/jboss-eap-6.2.0.zip")
  end

  it 'extract jboss' do
    expect(chef_run).to run_execute('jboss_extract')
  end

  it 'move directory to detination path' do
    expect(chef_run).to run_execute('jboss_install_path')
  end

  it 'change owner jboss' do
    expect(chef_run).to run_execute('jboss_owner')
  end

  it 'creates config directory for jboss' do
    expect(chef_run).to create_directory('/etc/jboss-as')
  end

  it 'creates pid directory for jboss' do
    expect(chef_run).to create_directory('/var/run/jboss-as')
  end

  it 'created jboss config file' do
    expect(chef_run).to create_template('/etc/jboss-as/app.conf')
  end

  it 'delete symlink to log jboss config file' do
    expect(chef_run).to delete_link('/var/log/jboss-as/app')
  end

  it 'creates log directory for jboss' do
    expect(chef_run).to create_directory('/var/log/jboss-as/app')
  end

  it 'delete default log directory for jboss' do
    expect(chef_run).to delete_directory('/opt/app/standalone/log')
  end

  it 'create symlink to log jboss config file' do
    expect(chef_run).to create_link('/opt/app/standalone/log')
  end

  it 'add admin user for jboss' do
    expect(chef_run).to run_execute('add_admin_user')
  end

  it 'declare jboss service' do
    execute = chef_run.service('app')
    expect(execute).to do_nothing
  end

  it 'create service script jboss' do
    expect(chef_run).to create_template('/etc/init.d/app')
  end

  it 'sends a notification to services' do
    expect(jboss_init).to notify('service[app]').to(:enable)
    expect(jboss_init).to notify('service[app]').to(:start)
  end

  it 'create standalone conf for jboss' do
    expect(chef_run).to create_template('/opt/app/bin/standalone.conf')
  end

  it 'sends a notification to services' do
    expect(jboss_standalone).to notify('service[app]').to(:restart)
  end

end


spec/centos/default_spec.rb
require_relative '../spec_helper'

describe 'jboss::default on Centos 6.5' do
  let(:chef_run) do
    ChefSpec::Runner.new(
      :platform => 'centos',
      :version => '6.5'
      ) do |node|
      node.set[:jboss][:version] = '6.2.0'
    end.converge('jboss::default')
  end

  let(:jboss_init) { chef_run.template('/etc/init.d/app') }
  let(:jboss_standalone) { chef_run.template('/opt/app/bin/standalone.conf') }

  before do
    stub_command('test -h /var/log/jboss-as/app').and_return(true)
    stub_command('test -h /opt/app/standalone/log').and_return(false)
    stub_command('grep admin /opt/app/standalone/configuration/mgmt-users.properties').and_return(false)
  end

  it 'installs a unzip package' do
    expect(chef_run).to install_package('install_unzip')
  end

  it 'creates a user for jboss' do
    expect(chef_run).to create_user('jboss')
  end

  it 'creates a group for jboss user' do
    expect(chef_run).to create_group('jboss')
  end

  it 'download a jboss package' do
    expect(chef_run).to create_remote_file("#{Chef::Config[:file_cache_path]}/jboss-eap-6.2.0.zip")
  end

  it 'extract jboss' do
    expect(chef_run).to run_execute('jboss_extract')
  end

  it 'move directory to detination path' do
    expect(chef_run).to run_execute('jboss_install_path')
  end

  it 'change owner jboss' do
    expect(chef_run).to run_execute('jboss_owner')
  end

  it 'creates config directory for jboss' do
    expect(chef_run).to create_directory('/etc/jboss-as')
  end

  it 'creates pid directory for jboss' do
    expect(chef_run).to create_directory('/var/run/jboss-as')
  end

  it 'created jboss config file' do
    expect(chef_run).to create_template('/etc/jboss-as/app.conf')
  end

  it 'delete symlink to log jboss config file' do
    expect(chef_run).to delete_link('/var/log/jboss-as/app')
  end

  it 'creates log directory for jboss' do
    expect(chef_run).to create_directory('/var/log/jboss-as/app')
  end

  it 'delete default log directory for jboss' do
    expect(chef_run).to delete_directory('/opt/app/standalone/log')
  end

  it 'create symlink to log jboss config file' do
    expect(chef_run).to create_link('/opt/app/standalone/log')
  end

  it 'add admin user for jboss' do
    expect(chef_run).to run_execute('add_admin_user')
  end

  it 'declare jboss service' do
    execute = chef_run.service('app')
    expect(execute).to do_nothing
  end

  it 'create service script jboss' do
    expect(chef_run).to create_template('/etc/init.d/app')
  end

  it 'sends a notification to services' do
    expect(jboss_init).to notify('service[app]').to(:enable)
    expect(jboss_init).to notify('service[app]').to(:start)
  end

  it 'create standalone conf for jboss' do
    expect(chef_run).to create_template('/opt/app/bin/standalone.conf')
  end

  it 'sends a notification to services' do
    expect(jboss_standalone).to notify('service[app]').to(:restart)
  end

end


recipes/jboss5.rb
# Encoding: utf-8
#
# Cookbook Name:: jboss
# Recipe:: jboss5
#
# Copyright 2014, Mariani Lucas
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

unless node.recipe?('jboss::default')
  Chef::Log.info('Using jboss::default instead is recommended.')
end

Chef::Log.info('Instalin Jboss 5 EAP.')

version_pkg = node[:jboss][:version].split('.').first(2).join('.')

package 'install_unzip' do
  package_name 'unzip'
  action :install
end

user node[:jboss][:jboss_user] do
  action :create
end

group node[:jboss][:jboss_group] do
  action :create
  members node[:jboss][:jboss_user]
end

remote_file "#{Chef::Config[:file_cache_path]}/#{node[:jboss][:filename]}.zip" do
  source node[:jboss][:package_url]
  checksum node[:jboss][:checksum]
  owner 'root'
  group 'root'
  mode 00755
  action :create
end

execute 'jboss_extract' do
  user 'root'
  cwd Chef::Config[:file_cache_path]
  command "unzip -o #{node[:jboss][:filename]}.zip"
  action :run
end

execute 'jboss_install_path' do
  user 'root'
  cwd Chef::Config[:file_cache_path]
  command "mv jboss-eap-#{version_pkg}/jboss-as #{node[:jboss][:install_path]}"
  action :run
  not_if { ::Dir.exist?("#{node[:jboss][:install_path]}/jboss-as") }
end

execute 'create_profile' do
  user 'root'
  cwd node[:jboss][:install_path]
  command "cp -R #{node[:jboss][:install_path]}/jboss-as/server/default #{node[:jboss][:install_path]}/jboss-as/server/#{node[:jboss][:application]}"
  action :run
  not_if { ::Dir.exist?("#{node[:jboss][:install_path]}/jboss-as/server/#{node[:jboss][:application]}") }
end

execute 'jboss_owner' do
  user 'root'
  cwd node[:jboss][:install_path]
  command "chown -R #{node[:jboss][:jboss_user]}. jboss-as"
  action :run
end

directory '/etc/jboss-as' do
  owner node[:jboss][:jboss_user]
  group node[:jboss][:jboss_group]
  mode '0755'
end

directory '/var/run/jboss-as' do
  owner node[:jboss][:jboss_user]
  group node[:jboss][:jboss_group]
  mode '0755'
end

template "#{node[:jboss][:install_path]}/jboss-as/server/#{node[:jboss][:application]}/conf/run.conf" do
  source 'jboss5-run.conf.erb'
  owner node[:jboss][:jboss_user]
  group node[:jboss][:jboss_group]
  mode '0755'
end

link "/etc/jboss-as/#{node[:jboss][:application]}" do
  to "#{node[:jboss][:install_path]}/jboss-as/server/#{node[:jboss][:application]}/conf"
  owner node[:jboss][:jboss_user]
  group node[:jboss][:jboss_group]
  not_if "test -h /etc/jboss-as/#{node[:jboss][:application]}"
end

link node[:jboss][:log_dir] do
  action :delete
  only_if "test -h #{node[:jboss][:log_dir]}"
end

directory node[:jboss][:log_dir] do
  recursive true
  owner node[:jboss][:jboss_user]
  group node[:jboss][:jboss_group]
  mode '0775'
end

directory "#{node[:jboss][:install_path]}/jboss-as/server/#{node[:jboss][:application]}/log" do
  recursive true
  action :delete
  not_if { node[:jboss][:log_dir] == "#{node[:jboss][:install_path]}/jboss-as/server/#{node[:jboss][:application]}/log" }
  not_if "test -h #{node[:jboss][:install_path]}/jboss-as/server/#{node[:jboss][:application]}/log"
end

link "#{node[:jboss][:install_path]}/jboss-as/server/#{node[:jboss][:application]}/log" do
  to node[:jboss][:log_dir]
  owner node[:jboss][:jboss_user]
  group node[:jboss][:jboss_group]
  not_if { node[:jboss][:log_dir] == "#{node[:jboss][:install_path]}/jboss-as/server//#{node[:jboss][:application]}/log" }
  not_if "test -h #{node[:jboss][:install_path]}/jboss-as/server/#{node[:jboss][:application]}/log"
end

service node[:jboss][:application] do
  supports :restart => true, :start => true, :stop => true, :reload => true
  action :nothing
end

template "/etc/init.d/#{node[:jboss][:application]}" do
  source 'jboss5-as-init.sh.erb'
  owner 'root'
  group 'root'
  mode '0755'
  notifies :enable, "service[#{node[:jboss][:application]}]"
  notifies :start, "service[#{node[:jboss][:application]}]"
end


recipes/jboss6.rb
# Encoding: utf-8
#
# Cookbook Name:: jboss
# Recipe:: jboss6
#
# Copyright 2014, Mariani Lucas
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

unless node.recipe?('jboss::default')
  Chef::Log.info('Using jboss::default instead is recommended.')
end

Chef::Log.info('Instalin Jboss 6 EAP.')

version_pkg = node[:jboss][:version].split('.').first(2).join('.')

package 'install_unzip' do
  package_name 'unzip'
  action :install
end

user node[:jboss][:jboss_user] do
  action :create
end

group node[:jboss][:jboss_group] do
  action :create
  members node[:jboss][:jboss_user]
end

remote_file "#{Chef::Config[:file_cache_path]}/#{node[:jboss][:filename]}.zip" do
  source node[:jboss][:package_url]
  checksum node[:jboss][:checksum]
  owner 'root'
  group 'root'
  mode 00755
  action :create
end

execute 'jboss_extract' do
  user 'root'
  cwd Chef::Config[:file_cache_path]
  command "unzip -o #{node[:jboss][:filename]}.zip"
  action :run
end

execute 'jboss_install_path' do
  user 'root'
  cwd Chef::Config[:file_cache_path]
  command "mv jboss-eap-#{version_pkg} #{node[:jboss][:install_path]}/#{node[:jboss][:application]}"
  action :run
  not_if { ::Dir.exist?("#{node[:jboss][:install_path]}/#{node[:jboss][:application]}") }
end

execute 'jboss_owner' do
  user 'root'
  cwd node[:jboss][:install_path]
  command "chown -R #{node[:jboss][:jboss_user]}. #{node[:jboss][:install_path]}/#{node[:jboss][:application]}"
  action :run
end

directory '/etc/jboss-as' do
  owner node[:jboss][:jboss_user]
  group node[:jboss][:jboss_group]
  mode '0755'
end

directory '/var/run/jboss-as' do
  owner node[:jboss][:jboss_user]
  group node[:jboss][:jboss_group]
  mode '0755'
end

template "/etc/jboss-as/#{node[:jboss][:application]}.conf" do
  source 'jboss-as.conf.erb'
  owner node[:jboss][:jboss_user]
  group node[:jboss][:jboss_group]
  mode '0755'
end

link node[:jboss][:log_dir] do
  action :delete
  only_if "test -h #{node[:jboss][:log_dir]}"
end

directory node[:jboss][:log_dir] do
  recursive true
  owner node[:jboss][:jboss_user]
  group node[:jboss][:jboss_group]
  mode '0775'
end

directory "#{node[:jboss][:install_path]}/#{node[:jboss][:application]}/standalone/log" do
  recursive true
  action :delete
  not_if { node[:jboss][:log_dir] == "#{node[:jboss][:install_path]}/#{node[:jboss][:application]}/standalone/log" }
  not_if "test -h #{node[:jboss][:install_path]}/#{node[:jboss][:application]}/standalone/log"
end

link "#{node[:jboss][:install_path]}/#{node[:jboss][:application]}/standalone/log" do
  to node[:jboss][:log_dir]
  owner node[:jboss][:jboss_user]
  group node[:jboss][:jboss_group]
  not_if { node[:jboss][:log_dir] == "#{node[:jboss][:install_path]}/#{node[:jboss][:application]}/standalone/log" }
  not_if "test -h #{node[:jboss][:install_path]}/#{node[:jboss][:application]}/standalone/log"
end

execute 'add_admin_user' do
  command "#{node[:jboss][:install_path]}/#{node[:jboss][:application]}/bin/add-user.sh --silent -u #{node[:jboss][:admin_user]} -p #{node[:jboss][:admin_passwd]}"
  not_if "grep #{node[:jboss][:admin_user]} #{node[:jboss][:install_path]}/#{node[:jboss][:application]}/standalone/configuration/mgmt-users.properties"
  not_if { node[:jboss][:admin_user] == '' }
  not_if { node[:jboss][:admin_passwd] == '' }
end

service node[:jboss][:application] do
  supports :restart => true, :start => true, :stop => true, :reload => true
  action :nothing
end

template "/etc/init.d/#{node[:jboss][:application]}" do
  source 'jboss-as-init.sh.erb'
  owner 'root'
  group 'root'
  mode '0755'
  notifies :enable, "service[#{node[:jboss][:application]}]"
  notifies :start, "service[#{node[:jboss][:application]}]"
end

template "#{node[:jboss][:install_path]}/#{node[:jboss][:application]}/bin/standalone.conf" do
  source 'jboss-as-standalone.conf.erb'
  owner 'root'
  group 'root'
  mode '0755'
  notifies :restart, "service[#{node[:jboss][:application]}]"
end
